### PR TITLE
fixup warnings

### PR DESCRIPTION
**Pull request by Febbe**

### CODE REVIEW

**Comment by Technius** [comment]
Can you break this into several PRs, each one focused on a specific set of files (e.g., each commit in this PR is a separate PR). I don't want to end up with a situation like in #3326 where the PR snowballed so much that it became unmanageable to review & merge. Also, this can introduce a lot of merge conflicts.

**Comment by Febbe** [comment]
The rest of the fixups here are either too much related or are very trivial changes. (e.g. rule of 5)

**Comment by Febbe** [comment]
Ignore clang-format, my version is clang-format 14, so differences are expected.

File: src/core/audio/AudioQueue.h [hunk 1]
```diff
@@ -121,9 +121,9 @@ class AudioQueue {
      * @return std::pair<double, int>,
```
Comment (Technius) [inline src/core/audio/AudioQueue.h:121]: Update this comment.

File: src/core/audio/VorbisConsumer.cpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ auto VorbisConsumer::start(const std::string& filename) -> bool {
 
     this->consumerThread = std::thread([this, sfFile = std::move(sfFile), channels = channels] {
         auto lock{audioQueue.acquire_lock()};
-        auto buffer_size{static_cast<size_t>(std::max(0, 64 * channels))};
+        auto buffer_size{size_t(64 * channels)};
```
Comment (Technius) [inline src/core/audio/VorbisConsumer.cpp:30]:
```suggestion
        size_t buffer_size(64 * channels);
```

File: src/core/pdf/base/XojCairoPdfExport.cpp [hunk 3]
```diff
@@ -160,15 +160,17 @@ auto XojCairoPdfExport::createPdf(fs::path const& file, PageRangeVector& range,
...
     if (this->progressListener) {
         this->progressListener->setMaximumState(count);
     }
 
     int c = 0;
-    for (PageRangeEntry* e: range) {
-        for (int i = e->getFirst(); i <= e->getLast(); i++) {
+    for (auto const& e: range) {
+        for (size_t i = e.getFirst(); i <= e.getLast(); i++) {
             if (i < 0 || i >= static_cast<int>(doc->getPageCount())) {
```
Comment (Technius) [inline src/core/pdf/base/XojCairoPdfExport.cpp:174]: Is this cast still needed?

File: src/util/PageRange.cpp [hunk 4]
```diff
@@ -3,29 +3,21 @@
...
-auto PageRangeEntry::getLast() const -> int { return this->last; }
-
-auto PageRangeEntry::getFirst() const -> int { return this->first; }
-
 auto PageRange::isSeparator(char c) -> bool { return (c == ',' || c == ';' || c == ':'); }
 
-auto PageRange::parse(const char* str, int pageCount) -> PageRangeVector {
+
+// Todo (rename): what does this? Neither name or code is explaining.
+auto PageRange::parse(std::string_view str, size_t pageCount) -> PageRangeVector {
```
Comment (Technius) [inline src/util/PageRange.cpp:10]: I believe this parses the string that's typed in the page range field of the export dialog. See `ExportDialog.cpp`.
image

Comment (Technius) [reply src/util/PageRange.cpp:10]: Ideally, there should be unit tests for this method.

File: src/core/audio/VorbisConsumer.cpp (hunk 2 above)
Comment (Febbe) [reply src/core/audio/VorbisConsumer.cpp:30]: This cast is actually required, since `64 * channels` is not size_t.

**Pull request merged**